tools/rpm2kmodkit

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env bash
+#
+# Create an archive of kernel development sources and toolchain.
+set -eu -o pipefail
+
+for opt in "$@"; do
+   optarg="$(expr "${opt}" : '[^=]*=\(.*\)')"
+   case "${opt}" in
+      --archive-dir=*) ARCHIVE_DIR="${optarg}" ;;
+      --toolchain-dir=*) TOOLCHAIN_DIR="${optarg}" ;;
+      --output-dir=*) OUTPUT_DIR="${optarg}" ;;
+   esac
+done
+
+# Use a friendly name for the top-level directory inside the archive.
+KMOD_KIT="${VARIANT}-${ARCH}-kmod-kit-v${VERSION_ID}"
+
+# Use the build ID within the filename, to align with our build's expectations.
+KMOD_KIT_FULL="bottlerocket-${VARIANT}-${ARCH}-${VERSION_ID}-${BUILD_ID}-kmod-kit"
+
+EXTRACT_DIR="$(mktemp -d)"
+KIT_DIR="$(mktemp -d)"
+mkdir -p "${OUTPUT_DIR}" "${KIT_DIR}/${KMOD_KIT}"
+
+# Extract any RPMs and find the kernel development archive.
+pushd "${EXTRACT_DIR}" >/dev/null
+find "${ARCHIVE_DIR}" -type f -name '*.rpm' \
+  -exec rpm2cpio {} \; | cpio -idmu --quiet
```

Comment:
Do we need the `-u`?  Are there cases where we have to replace files from prior RPMs?  That seems like a problem waiting to happen, and it could be difficult to debug if file contents mysteriously change from what you expect.  (I know it was there before, so call it a nit, but we're adding more RPMs...)